HW2/P2/parallel_vector.pyx

```diff
@@ -107,9 +113,15 @@ cpdef move_data_medium_grained(np.int32_t[:] counts,
    ##########
    with nogil:
        for r in range(repeat):
-           for idx in range(src.shape[0]):
+           #for idx in range(src.shape[0]):
+           for idx in prange(src.shape[0], num_threads=4):
                if counts[src[idx]] > 0:
+                   acquire(&locks[dest[idx]/N])
                    counts[dest[idx]] += 1
+                   release(&locks[dest[idx]/N])
+
+                   acquire(&locks[src[idx]/N])
                    counts[src[idx]] -= 1
+                   release(&locks[src[idx]/N])
 
```

Comment:
I did the same thing- lock the count and immediately release it after updating. I think other people locked the two updates together.